### ISSUE TITLE
Could objective function be a pytorch model?

### ISSUE THREAD

**Issue by sakuraiiiii** [cvxgrp/cvxpylayers#68]
Hi. I have a question that could the objective function be a pytorch model? For example, I define a pytorch model and want to find the optimal input which makes the output of the model is minimum.
```python
import torch
import cvxpy as cp
from cvxpylayers.torch import Cvxpylayer

class net(nn.Module):
    def __init__(self):
        super(net, self).__init__()
        self.lin = nn.Sequential(nn.Linear(20, 100), nn.ReLU(), nn.Linear(100, 1))

    def forward(self, x):
        return self.lin(x)

x = cp.Parameter(20)
model = net()
objective = cp.Minimize(model(x))
constraints = [x > 0]
problem = cp.Problem(objective, constraints)
cvxpylayer = CvxpyLayer(problem, parameters=[x], variables=[model.parameters()])
...
```
The above code is not right, but that's what I mean. I have no idea about how to correct it. Could you give me some advice? Thank you.

**Comment by bamos** [comment]
Hi, because the network is generally non-convex CVXPY isn't the best fit in this context. One potential alternative is to do (potentially projected) gradient steps over the input space, like in https://arxiv.org/abs/1707.05776. I'll close this issue for now but let us know if there's anything else!

**Comment by sakuraiiiii** [comment]
Yes, the network is usually non-convex. But I think CVXPY could still find some local minimum. I want to find what the result will happen. Thus how to implenment the objective function as a pytorch model?

**Comment by bamos** [comment]
The challenge with using a PyTorch module in the CVXPY objective is that CVXPY needs a provably convex analytic form in the objective expressed in CVXPY atoms and it can't call into arbitrary Python or PyTorch modules. You could potentially using something like [this](https://arxiv.org/abs/1811.01988) but I still think not using CVXPY and doing projected gradient steps on the PyTorch module directly is more promising

**Comment by sakuraiiiii** [comment]
Thank you very much.
